lib/queue.js

```diff
@@ -279,6 +279,12 @@ function redisClientGetter(queue, options, initCallback) {
         return connections[type];
       }
       const client = (connections[type] = createClient(type, options.redis));
+
+      // Since connections are lazily initialized, we can't check queue.client
+      // without initializing a connection. So expose a boolean we can safely
+      // query.
+      queue[type + 'Initialized'] = true;
```

Comment:
Couldn't you store this information in a separate data structure? furthermore, what other queue types would benefit from storing this data?

Reply:
> Couldn't you store this information in a separate data structure?

Not sure I follow. Do you mean you want a dedicated object on the queue to save this? Something like
```js
queue._clientStatuses = {}; // in Queue constructor

queue._clientStatuses[type] = 'initialized'; // after initialization
```

Or are you thinking there's a more complex state machine that should be used?

> what other queue types would benefit from storing this data?

Do you mean Redis connection types? There may be instances where `client` and `eclient` are also initialized unnecessarily, but I haven't tracked them down yet. That's also blocked by #1344 which means that `client` is always initialized.

Reply:
@manast Have you had a chance to look at this?

Reply:
yes, sorry. I meant a separate data structure.